client/ayon_core/hosts/max/plugins/publish/collect_workfile.py

```diff
@@ -23,15 +24,12 @@ def process(self, context):
 
         context.data['currentFile'] = current_file
 
-        filename, ext = os.path.splitext(file)
-
-        task = context.data["task"]
+        ext = os.path.splitext(file)[-1].lstrip(".")
 
         data = {}
 
         # create instance
-        instance = context.create_instance(name=filename)
-        subset = 'workfile' + task.capitalize()
+        subset = instance.data["subset"]
 
         data.update({
             "subset": subset,
```

Comment:
The issue described:

> For some reason, get_subset_name function can't set the name with the correct variant, although there is variant.
> It is using the old subset name workfileAsset.

Is likely due to this setting `subset`?

Should be removed here.